datatrails-common-api/assets/v2/assets/merklelogentry.proto

```diff
@@ -8,12 +8,32 @@ import "google/protobuf/timestamp.proto";
 // MerkeLogCommit provides the log entry details for a single mmr leaf.
 message MerkleLogCommit {
     /* The mmr index */
-    uint64 index = 3;
+    uint64 index = 1;
     /* The mmr *leaf* index */
-    uint64 leaf_index = 4; // TBD: this may be redundant.
+    uint64 leaf_index = 2; // TBD: this may be redundant.
     /* time ordered and strictly unique per tenant. system wide
      * unique with very reasonable operational assumptions. */
-    fixed64 idtimestamp = 5;
+    fixed64 idtimestamp = 3;
+}
+
+message MerkleLogCommitMongoDB {
+
+    // Note that should we ever have more than 2^63  events in a tenant log, the
+    // index and leaf_index fields will not persist to mongo.
+
+    /* The mmr index */
+    uint64 index = 1;
+    /* The mmr *leaf* index */
+    uint64 leaf_index = 2; // TBD: this may be redundant.
+
+    /* time ordered and strictly unique per tenant. system wide
+     * unique with very reasonable operational assumptions.
+     *
+     * EXPRESSED AS A 16 character padded hex string because mongo db does not
```

Comment:
Same applies here, this i believe will be exposed in the swagger documentation, we shouldn't mention our implementation constraints to customers?

Reply:
see resolution above.

Reply:
I think the trick for the verbose commentary is to leave a blank line

Reply:
ah no, you are right. it makes it throug :-/ I will remove it in a followup

Reply:
[User Story 8987](https://dev.azure.com/jitsuin/avid/_workitems/edit/8987): Tidy up forestrie related swagger comments